src/pages/SetNewPasswordPage.tsx

```diff
@@ -0,0 +1,73 @@
+import { Formik, Form, Field } from 'formik';
+import { Link } from "react-router-dom";
+import { passwordRegex } from "../validationRules";
+import "../components/modal/modal.scss";
+import PasswordAndConfirm from "../components/PasswordAndConfirm";
+import { PASSWORD } from "../constants";
+
+interface FormValues {
+  password: string;
+  confirmPassword: string;
+}
+
+interface FormErrors {
+  [key: string]: string
+}
+
+const SetNewPasswordPage = () => {
+
+  const initialValues: FormValues = {
+    password: '',
+    confirmPassword: '',
+  }
+
+  const validate = async (values: FormValues) => {
+    let errors: FormErrors = {};
+    if (!passwordRegex.test(values.password)) {
+      errors.password = `An invalid character is present in the password. Password must be between ${PASSWORD.minLength} and ${PASSWORD.maxLength} characters; upper or lower case Latin letters (a–z, A–Z); numbers from 0 to 9; symbols ! # $ % & ' * + - / = ? ^ _ \` { | } ~`;
+    }
+    if (values.password.length >= PASSWORD.maxLength || values.password.length < PASSWORD.minLength) {
```

Comment:
Maybe 

if (values.password.length > PASSWORD.maxLength || values.password.length < PASSWORD.minLength) ?

">" instead of ">=" ?